src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +34,109 @@ public PropertySubpatternCompletionProvider()
         {
         }
 
+        // Examples:
+        // is { $$
+        // is { Property.$$
+        // is { Property.Property2.$$
```

Comment:
ok.  we may need to chat.  but i just realized something... why do we need any special logic here at all.  So the old code took care of hte case of typing just an identifier into a property.  I get why we needed that.  However, once you type `X.` then you're firmly in expression land, in which case why do we need to anything special.  Shouldn't normal symbol-completion just work here?  Could you clarify why we need any of this code in that acse?  Thanks!

Reply:
Sure. Let's chat tomorrow (I was out today).

The binding for `c is { X. }` doesn't involve the type of `c`. Consider this as a list-pattern: `c is { X.Constant }`, the meaning of `X.Constant` doesn't depend on the type of `c` (a property `C.X` would not bind). But in a property-pattern, the property has an implicit receiver (the "input type" of the property pattern).
The point of `PropertySubPatternCompletionProvider` is to provide completion off of the type `C`, and now also off of the type of `C.X` or `C.X.Y`, etc.

Reply:
got itttttttttt.  that makes a ton of sense :)   so yeah, def add this as a comment to the code :)